### DISCUSSION TITLE
Protocol inheritance doesn't work on lib

### DISCUSSION THREAD

**Discussion by IgorMaineti** [vinivendra/Gryphon#99] [General]
Recently I opened a pull request to implement protocol inheritance #95 and it worked fine on the test class, however, after the merge I installed the HEAD version from brew and it didn't work 😞.

The code I wrote on the test class:

`public protocol DisplayComponent : Component {
}`

Output:

`interface DisplayComponent: Component {
}`

But, if I write the same protocol on my lib, gryphon won't translate their inheritances, could you help us, please?

**Comment by vinivendra** [comment]
Sure, let's look into it. If I understand you, Gryphon is translating something like `interface DisplayComponent { }`, right? If that's it, we need to find out where the inheritance is being removed. Can you add a `gryphon inspect` comment before the protocol?

```` swift
// gryphon inspect
public protocol DisplayComponent : Component { }
````

When you run it with the comment, Gryphon should print some information about this protocol to the command line. Then, you can paste that information here and I'll take a look at it.

If you're using Gryphon through Xcode (instead of the command line) you can probably see the output by hitting Command+9 and selecting the latest `Gryphon` build.

**Comment by vinivendra** [comment]
Hey @IgorMaineti, any news on this? Did it work out?

**Comment by IgorMaineti** [comment]
Hello @vinivendra, sorry for the late answer, the holidays came and I totally forgot about this discussion. We observed a strange behaviour, the transpilation worked fine when we ran Gryphon through Xcode but it wasn't working building through the terminal, however, yesterday we all updated everything again and it worked flawslessly. Thanks for everything, you can close the discussion 😄 .

**Comment by vinivendra** [comment]
No worries, just wanted to make sure everything was OK :)
